tests/Umbraco.Tests.Benchmarks/GuidUdiBenchmarks.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using BenchmarkDotNet.Attributes;
+using Umbraco.Cms.Core;
+using Umbraco.Tests.Benchmarks.Config;
+
+namespace Umbraco.Tests.Benchmarks
+{
+    [QuickRunWithMemoryDiagnoserConfig]
+    public class GuidUdiBenchmarks
+    {
+        private readonly Guid _guid = Guid.NewGuid();
+        private readonly string _entityType = Constants.UdiEntityType.DocumentType;
+
+        [Benchmark(Baseline = true)]
+        public Udi CurrentInit()
+        {
+            return new OldGuidUdi(_entityType, _guid);
+        }
+
+        [Benchmark]
+        public Udi NewInit()
+        {
+            return new NewGuidUdi(_entityType, _guid);
+        }
+
+        public class OldGuidUdi : Udi
+        {
+            public Guid Guid { get; }
+
+            public override bool IsRoot => throw new NotImplementedException();
+
+            public OldGuidUdi(string entityType, Guid guid)
+            : base(entityType, "umb://" + entityType + "/" + guid.ToString("N")) =>
+            Guid = guid;
+        }
+
+        public class NewGuidUdi : Udi
+        {
+            public Guid Guid { get; }
+
+            public override bool IsRoot => throw new NotImplementedException();
+
+            public NewGuidUdi(string entityType, Guid guid) : base(entityType, GetStringValue(entityType, guid))
+            {
+
```

Comment:
Hey @patrickdemooij9 , I really don't know the impact on the benchmark,  but in the `OldGuidUdi `you assign the Guid property at the end of the constructor, so I would add the assignment of the guid to the Guid property somewhere in here as well, to be sure we benchmark the same behavior.

Reply:
Good point, while it shouldn't have any impact on the allocated memory as the guid is already allocated before the benchmarks, it is important to keep that the same

Reply:
Added it, doesn't seem to have any big impact on the benchmark